### ISSUE TITLE
Document route matching priority

### ISSUE THREAD

**Issue by evanworley** [visionmedia/page.js#137]
Is there any documentation anywhere which describes how routes are matched when multiple patterns match a given route. For instance

/projects/:id vs /projects/new. Both routes match the url "/projects/new". Does order of registration matter, or is there a hierarchy of match qualities?

**Comment by tractorcow** [comment]
A problem to consider is that not all routes may be bound up front; Components (such as short-lived edit forms for a backend model) may be loaded in via ajax, which may have specific routes attached to them. It may not always be possible to pre-determine all routes in such cases, so the problem still exists here.

Possible solutions I've toyed with:

- Hack page.js. e.g. implement some kind of page.callbacks.unshift(), or just resetting page.callbacks to an empty array and re-registering. It's not very robust to rely on specific internal workings, however.

- Add some complex code to less-specific routes, such that they are aware of other routes further on down the next() queue, and do nothing if another callback is expected to handle any request.

- Enjoy more non-ajax full page reloads and accept that you can't fully route an entire application.

**Comment by tractorcow** [comment]
Right, so one solution is here; https://github.com/visionmedia/page.js/issues/371
